src/Workspaces/Remote/ServiceHub/Host/RemoteDocumentDifferenceService.cs

```diff
@@ -43,35 +43,9 @@ public VisualBasicDocumentDifferenceService()
             }
         }
 
-        public async Task<DocumentDifferenceResult?> GetDifferenceAsync(Document oldDocument, Document newDocument, CancellationToken cancellationToken)
+        public Task<SyntaxNode?> GetChangedMemberAsync(Document oldDocument, Document newDocument, CancellationToken cancellationToken)
         {
-            // in remote workspace, we don't trust any version based on VersionStamp. we only trust content based information such as
-            // checksum or tree comparison and etc.
-
-            // first check checksum
-            var oldTextChecksum = (await oldDocument.State.GetStateChecksumsAsync(cancellationToken).ConfigureAwait(false)).Text;
-            var newTextChecksum = (await newDocument.State.GetStateChecksumsAsync(cancellationToken).ConfigureAwait(false)).Text;
-            if (oldTextChecksum == newTextChecksum)
-            {
-                // null means nothing has changed.
-                return null;
-            }
-
-            var oldRoot = await oldDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var newRoot = await newDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-
-            // the service is only registered for C# and VB documents, which must have syntax trees:
-            Contract.ThrowIfNull(oldRoot);
-            Contract.ThrowIfNull(newRoot);
-
-            if (oldRoot.IsEquivalentTo(newRoot, topLevel: true))
-            {
-                // only method body changed
-                return new DocumentDifferenceResult(InvocationReasons.SyntaxChanged);
-            }
-
-            // semantic has changed as well.
-            return new DocumentDifferenceResult(InvocationReasons.DocumentChanged);
+            return SpecializedTasks.Null<SyntaxNode>();
```

Comment:
the old code never actually returned a syntax node.  only the type of change it observed.  this preserves that logic (of just returning 'null' as the changed member).